### PR TITLE
Add cmake find_package compatibility

### PR DESCRIPTION
**Pull request by ascii255**
mainly following the cmake [Importing and Exporting Guide](https://cmake.org/cmake/help/latest/guide/importing-exporting/index.html?highlight=cmakepackageconfighelpers#id1)

### CODE REVIEW

**Comment by ascii255** [comment]
bump, can somebody have a look at this?

File: CMakeLists.txt [hunk 1]
```diff
@@ -201,16 +203,54 @@ set(CC_SOURCES
...
 add_library(sptk STATIC ${CC_SOURCES})
 target_include_directories(sptk PUBLIC
-  ${PROJECT_SOURCE_DIR}/include
-  ${THIRD_PARTY_DIR}
-  ${THIRD_PARTY_DIR}/REAPER
-  ${THIRD_PARTY_DIR}/WORLD
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}/REAPER>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}/WORLD>
```
Comment (takenori-y) [inline CMakeLists.txt:209]: Do you need to add <INSTALL_INTERFACE>?

File: CMakeLists.txt [hunk 2]
```diff
@@ -201,16 +203,54 @@ set(CC_SOURCES
...
+  )
+
+install(EXPORT ${PROJECT_NAME}Targets
+  FILE ${PROJECT_NAME}Targets.cmake
+  NAMESPACE ${PROJECT_NAME}::
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+  )
+
+file(WRITE ${PROJECT_BINARY_DIR}/Config.cmake.in
+  "@PACKAGE_INIT@\n\ninclude(\"\${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake\")\n\ncheck_required_components(@PROJECT_NAME@)\n"
```
Comment (takenori-y) [inline CMakeLists.txt:234]: Is it normal to generate Config.cmake.in? Should it put on the repository?

File: CMakeLists.txt [hunk 3]
```diff
@@ -201,16 +203,54 @@ set(CC_SOURCES
...
+  ${PROJECT_BINARY_DIR}/Config.cmake.in
+  ${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
+  INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+  )
+
+write_basic_package_version_file(
+  ${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
+  VERSION ${PROJECT_VERSION}
+  COMPATIBILITY SameMajorVersion
+  ARCH_INDEPENDENT
```
Comment (takenori-y) [inline CMakeLists.txt:247]:
```suggestion
```
since old cmake versions do not support this.

**Review by takenori-y — CHANGES_REQUESTED** [review]
Sorry for the delay response since I am not familiar with cmake.

File: CMakeLists.txt (hunk 1 above)
Comment (ascii255) [reply CMakeLists.txt:209]: This is necessary. The include path is only valid, when running cmake from the source dir. When installed, the include path should be something like /usr/local/include. Also, if you leave it out, you will get following errors when installing:

  Target "sptk" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "/Users/thomas/Developer/sptk/include"

  which is prefixed in the source directory.

File: CMakeLists.txt (hunk 2 above)
Comment (ascii255) [reply CMakeLists.txt:234]: I can put it in the repository as a file as well. It makes no difference. Do want another file in the repo?

File: CMakeLists.txt [hunk 4]
```diff
@@ -14,9 +14,11 @@
 # limitations under the License.                                           #
 # ------------------------------------------------------------------------ #
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.14)
 
-project(SPTK)
+project(SPTK
+  VERSION 4.1.0
```
[the remark this answers is not in the document]
Comment (ascii255) [reply CMakeLists.txt:20]: will do

File: CMakeLists.txt (hunk 2 above)
Comment (takenori-y) [reply CMakeLists.txt]: Thank you for changing.

File: CMakeLists.txt (hunk 1 above)
Comment (takenori-y) [reply CMakeLists.txt:209]: Sorry, my question is not appropriate. You added `BUILD_INTERFACE` lines but not `INSTALL_INTERFACE`lines.

Do we need to add the following line?
```
$<INSTALL_INTERFACE:include>
```

**Comment by takenori-y** [comment]
I added a windows test to the master branch. Could you merge it? Then I will run the test.

File: CMakeLists.txt (hunk 1 above)
Comment (ascii255) [reply CMakeLists.txt:209]: I don't think so. There is this line in the file
`INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}`
This was always enough for me to let cmake do the right thing when installing.

**Comment by ascii255** [comment]
> I added a windows test to the master branch. Could you merge it? Then I will run the test.

I merged it

File: CMakeLists.txt (hunk 1 above)
Comment (ascii255) [reply CMakeLists.txt:213]: > INCLUDES DESTINATION
This option specifies a list of directories which will be added to the [INTERFACE_INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html#prop_tgt:INTERFACE_INCLUDE_DIRECTORIES) target property of the <targets> when exported by the [install(EXPORT)](https://cmake.org/cmake/help/latest/command/install.html#install-export) command. If a relative path is specified, it is treated as relative to the [$<INSTALL_PREFIX>](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:INSTALL_PREFIX).
source: [https://cmake.org/cmake/help/latest/command/install.html](https://cmake.org/cmake/help/latest/command/install.html)

**Review by takenori-y — APPROVED** [review]
LGTM. Thank you for your interaction.

**Pull request merged**